src/domain/dataverse/domain.ts

```diff
@@ -179,7 +199,25 @@ export const storeFactory = (
                 )
               )
             )
-          )
+          ),
+        setByPropertyFilter: (newFilter: ByPropertyFilterInput) => () =>
+          set(state => ({
+            data: {
+              ...state.data,
+              dataverse: [],
+              filters: {
+                ...state.data.filters,
+                byProperty: pipe(
+                  newFilter.value,
+                  O.fromPredicate(v => !!v.length),
+                  O.match(
+                    () => null,
+                    () => newFilter
+                  )
+                )
```

Comment:
Option shouldn't be used for conditional branching, use [B.match](https://gcanti.github.io/fp-ts/modules/boolean.ts.html#match) instead; the semantic is clearer and more expressive, imho.

I think the following should do the job:

```suggestion
                byProperty: pipe(
                  !!newFilter.value.length,
                  B.match(
                    () => null,
                    () => newFilter
                  )
                )
```

Reply:
much better, thanks !!!